includes/helper-functions.php

```diff
@@ -0,0 +1,305 @@
+			}
+			if ( strpos( $key, 'font_ttf-' ) !== false && $value ) {
+				$consider_counter                  = true;
+				$temp_arr[ $counter ]['font_url']  = esc_url( $value );
+				$temp_arr[ $counter ]['font_file'] = bcf_get_media_image_id_by_url( $value );
+			}
+			if ( strpos( $key, 'font_eot-' ) !== false && $value ) {
+				$consider_counter                  = true;
+				$temp_arr[ $counter ]['font_url']  = esc_url( $value );
+				$temp_arr[ $counter ]['font_file'] = bcf_get_media_image_id_by_url( $value );
+			}
+			if ( strpos( $key, 'font_svg-' ) !== false && $value ) {
+				$consider_counter                  = true;
+				$temp_arr[ $counter ]['font_url']  = esc_url( $value );
+				$temp_arr[ $counter ]['font_file'] = bcf_get_media_image_id_by_url( $value );
+			}
+			if ( strpos( $key, 'font_otf-' ) !== false && $value ) {
+				$consider_counter                  = true;
+				$temp_arr[ $counter ]['font_url']  = esc_url( $value );
+				$temp_arr[ $counter ]['font_file'] = bcf_get_media_image_id_by_url( $value );
+			}
+
+			if ( true === $consider_counter ) {
+				$temp_arr[ $counter ]['id']         = $counter + 1;
+				$temp_arr[ $counter ]['font_style'] = 'normal';
+			}
+		}
+
+		$font_data['variations'] = $temp_arr;
+	}
+
+	return $font_data;
+}
+
+/**
+ * Sanitize data recursively, eg: font-face data.
+ *
+ * @since x.x.x
+ */
+function bcf_sanitize_text_field_recursive( $data ) {
+	if ( is_array( $data ) ) {
+		foreach ( $data as $key => $value ) {
+			$data[ $key ] = bcf_sanitize_text_field_recursive( $value );
+		}
+
+		return $data;
+	}
+	return sanitize_text_field( $data );
+}
+
+/**
+ * Get font source as per the asked type.
+ *
+ * @since x.x.x
+ */
+function bcf_get_font_src( $font_type, $font_url ) {
+	$font_src = 'url(\'' . esc_url( $font_url ) . '\') ';
+	switch ( $font_type ) {
+		case 'woff':
+		case 'woff2':
+		case 'svg':
+			$font_src .= 'format(\'' . $font_type . '\')';
+			break;
+
+		case 'ttf':
+			$font_src .= 'format(\'truetype\')';
+			break;
+
+		case 'otf':
+			$font_src .= 'format(\'OpenType\')';
+			break;
+
+		case 'eot':
+			$font_src = 'url(\'' . esc_url( $font_url ) . '?#iefix\') format(\'embedded-opentype\')';
+			break;
+	}
+
+	return $font_src;
+}
+
+/**
+ * Get the file extension of font file.
+ *
+ * @return mixed
+ * @since x.x.x
+ */
+function bcf_get_font_file_extension( $font_url ) {
+	$file_last_chunk = substr( $font_url, -6 );
+	if ( strpos( $file_last_chunk, '.woff2' ) !== false ) {
+		return 'woff2';
+	}
+
+	if ( strpos( $file_last_chunk, '.woff' ) !== false ) {
+		return 'woff';
+	}
+
+	if ( strpos( $file_last_chunk, '.ttf' ) !== false ) {
+		return 'ttf';
+	}
+
+	if ( strpos( $file_last_chunk, '.eot' ) !== false ) {
+		return 'eot';
+	}
+
+	if ( strpos( $file_last_chunk, '.svg' ) !== false ) {
+		return 'svg';
+	}
+
+	if ( strpos( $file_last_chunk, '.otf' ) !== false ) {
+		return 'otf';
+	}
+
+	return false;
+}
+
+/**
+ * Based on the font post data create font-face CSS.
+ *
```

Comment:
@params are not added to the function @imnavanath

Reply:
DONE